M2/Macaulay2/m2/ringmap.m2

```diff
@@ -435,17 +435,10 @@ RingMap ** Matrix := Matrix => (f,m) -> (
      if source f =!= ring m then error "expected matrix over source ring";
      map(f ** target m, f ** source m, f cover m))
 
-tensor(Ring,RingMap,Module) := opts -> (S,f,M) -> (
-     if S =!= target f then error "tensor: expected ring and target of ring map to be the same";
-     f ** M)
-
-tensor(Ring,RingMap,Matrix) := opts -> (S,f,m) -> (
-     if S =!= target f then error "tensor: expected ring and target of ring map to be the same";
-     f ** m)
-
-tensor(RingMap,Module) := Module => opts -> (f,M) -> f ** M
-
-tensor(RingMap,Matrix) := Matrix => opts -> (f,m) -> f ** m
+tensor(Ring, RingMap) := RingMap => {} >> opts -> (S, f) -> (
```

Comment:
This method doesn't seem mathematical, and it doesn't do anything, either.  What's it for?

Reply:
This is to allow for the [method](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_tensor_lp__Ring_cm__Ring__Map_cm__Matrix_rp.html) `(tensor, Ring, RingMap, Matrix)`.

I think it makes as much sense as `(tensor, RingMap, Matrix)`, don't you think?

Reply:
> This is to allow for the [method](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_tensor_lp__Ring_cm__Ring__Map_cm__Matrix_rp.html) `(tensor, Ring, RingMap, Matrix)`.
> 

In what sense does it "allow for" the [method](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_tensor_lp__Ring_cm__Ring__Map_cm__Matrix_rp.html) `(tensor, Ring, RingMap, Matrix)`.

> I think it makes as much mathematical sense as `(tensor, RingMap, Matrix)`, don't you think?

No, since it never does any mathematical computation, and it doesn't correspond to any mathematical operation.

Reply:
> In what sense does it "allow for" the [method](https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_tensor_lp__Ring_cm__Ring__Map_cm__Matrix_rp.html) `(tensor, Ring, RingMap, Matrix)`.

I think this comment answers your question:
https://github.com/Macaulay2/M2/blob/8950020fd3018ed094e38cedb8738be5fede37a4/M2/Macaulay2/m2/intersect.m2#L75-L80

> No, since it never does any mathematical computation, and it doesn't correspond to any mathematical operation.

Okay, then let's get rid of this and `(tensor, Ring, RingMap, Matrix)` at the same time, because the latter has a redundant Ring input.

Reply:
Okay, sure.

Reply:
Done.